tool/tsh/common/app_azure_test.go

```diff
@@ -79,136 +79,136 @@ func TestAzure(t *testing.T) {
+		}
+		return ""
+	}
+
+	versionWithoutMSAL := semver.New(azureCLIVersionMSALRequirement.String())
+	versionWithoutMSAL.Minor -= 1
+
+	for name, tc := range map[string]struct {
+		setEnvironment       func(t *testing.T)
+		cliVersion           *semver.Version
+		tokenEndpointURL     string
+		expectedLoginCommand []string
+		assertCommandEnv     require.ValueAssertionFunc
 	}{
-		{
-			name:         "incomplete request",
-			url:          "https://azure-msi.teleport.dev/very-secret",
-			headers:      nil,
-			expectedCode: 400,
-			expectedBody: []byte("{\n    \"error\": {\n        \"message\": \"expected Metadata header with value 'true'\"\n    }\n}"),
+		"MSI": {
+			setEnvironment: func(t *testing.T) {
+				// This is required to avoid having a random generated secret.
+				t.Setenv("MSI_ENDPOINT", "https://azure-msi.teleport.dev/very-secret")
+			},
+			cliVersion:           versionWithoutMSAL,
+			tokenEndpointURL:     "https://azure-msi.teleport.dev/very-secret",
+			expectedLoginCommand: []string{"az", "login", "--identity", "-u", "dummy_azure_identity"},
+			assertCommandEnv: func(t require.TestingT, val any, msgAndArgs ...any) {
+				env := val.([]string)
+				require.Equal(t, "https://azure-msi.teleport.dev/very-secret", getEnvValue(env, "MSI_ENDPOINT"))
+			},
 		},
-		{
-			name:         "well-formatted request",
-			url:          "https://azure-msi.teleport.dev/very-secret?resource=myresource&msi_res_id=dummy_azure_identity",
-			headers:      map[string]string{"Metadata": "true"},
-			expectedCode: 200,
-			verifyBody: func(t *testing.T, body []byte) {
-				var req struct {
-					AccessToken  string `json:"access_token"`
-					ClientID     string `json:"client_id"`
-					Resource     string `json:"resource"`
-					TokenType    string `json:"token_type"`
-					ExpiresIn    int    `json:"expires_in"`
-					ExpiresOn    int    `json:"expires_on"`
-					ExtExpiresIn int    `json:"ext_expires_in"`
-					NotBefore    int    `json:"not_before"`
+		"Identity": {
+			setEnvironment: func(t *testing.T) {
+				// This is required to avoid having a random generated secret.
+				t.Setenv("IDENTITY_ENDPOINT", "https://azure-identity.teleport.dev")
+				t.Setenv("IDENTITY_SECRET", "very-secret")
+			},
+			cliVersion:           azureCLIVersionMSALRequirement,
+			tokenEndpointURL:     "https://azure-identity.teleport.dev",
+			expectedLoginCommand: []string{"az", "login", "--identity", "--resource-id", "dummy_azure_identity"},
+			assertCommandEnv: func(t require.TestingT, val any, msgAndArgs ...any) {
+				env := val.([]string)
+				require.Equal(t, "https://azure-identity.teleport.dev", getEnvValue(env, "IDENTITY_ENDPOINT"))
+				require.Equal(t, "very-secret", getEnvValue(env, "IDENTITY_SECRET"))
+			},
+		},
+	} {
+		t.Run("With"+name, func(t *testing.T) {
+			handleAzVersion := func(cmd *exec.Cmd) bool {
+				if len(cmd.Args) > 0 && cmd.Args[1] == "version" {
+					fmt.Fprintf(cmd.Stdout, `{"azure-cli-core": "%s"}`, tc.cliVersion.String())
```

Comment:
nit: we only need azure-cli-core now but the regular output is small enough i think we should include a sample here for reference purpose.